### DISCUSSION TITLE
Debugging?

### DISCUSSION THREAD

**Discussion by KungPhoo** [drmortalwombat/oscar64#55] [Q&A] [answered]
Is it possible to debug with an emulator, so I can have breakpoints and read variable values? If so - how would I do this?

**Comment by drmortalwombat** [comment, marked as answer]
I can recommend two options.  
1.) Use vice and load the compiler generated lable file with the -moncommands:
`C:\C64\GTK3VICE-3.8-win64\bin\x64sc -moncommands ninia.lbl ninia.crt `
keep the .asm file around to look up what happens.  Using the -g option in the compiler will generate a more annotated .asm file.
2.) Use a source level debugger such as [https://github.com/MihaMarkic/modern-vice-pdb-monitor](url) which has source level support for oscar
